### PR TITLE
feat(prettier): Add Prettier formatter

### LINKED ISSUE #26: Use Prettier for code formatting and linting
Prettier automatically formats code, which ensures consistent code
format across the codebase. This is highly simplifies code reviews and
limits the amount of core format nits in reviews. This is especially valuable in projects with large number of contributors.
    
Unfortunatelly Prettier does not work by default with standardjs so to
get the best of both worlds we can use eslint configs used by standardjs and combine them with prettier linting.

### PR DESCRIPTION
**Pull request by jtomasek**
**What**:
Prettier automatically formats code, which ensures consistent code
format across the codebase.

**Why**:
Prettier takes care of code formatting so reviewer does not have to focus on this aspect of reviews. Speeds up review as well as amount of updates to PRs

**How**:
Unfortunatelly Prettier does not work by default with standardjs so to
get the best of both worlds this adds custom eslint config which uses
standardjs config and prettier config which ensures that standardjs
rules which conflict with prettier are removed. In addition
eslint-plugin-prettier adds Prettier rules for linting.

To prettify code run 'npm run prettier' or use prettier plugin in editor
of your choice.

Closes #26

### CODE REVIEW

File: src/Alert/Alert.js [hunk 1]
```diff
@@ -24,21 +24,27 @@ const Alert = ({ children, onDismiss, type }) => {
   return (
     <div className={alertClass}>
       {onDismiss &&
-        <button type='button' className='close' aria-hidden='true' onClick={onDismiss}>
-          <span className='pficon pficon-close' />
+        <button
+          type="button"
```
Comment (priley86) [inline src/Alert/Alert.js]: hey @jtomasek, the only distinction i'm seeing here in my VS Code editor is my old standardjs plugin now complains about double quote in the JSX attributes. I'm fine w/ disabling this if we prefer to use double quotes now. Is this consistent w/ your eslint config?

**Comment by priley86** [comment]
great job w/ this, I really like Prettier... 👍

File: src/Alert/Alert.js (hunk 1 above)
Comment (priley86) [inline src/Alert/Alert.js]: i think we are safe to merge this...just wanted the clarification on it before I go disable that old plugin ;)

**Comment by ohadlevy** [comment]
nice idea :) /cc @tbrisker and @matanwerbner

File: src/Alert/Alert.js (hunk 1 above)
Comment (jtomasek) [inline src/Alert/Alert.js]: This rule is one of the collisions between prettier and srandardjs. eslint-config-prettier is taking care of this.  You'll need to disable your standardjs plugin in your editor and use eslint based plugin. Technically we're not using standardjs any more, we're just using it's eslint config.

Personally I very much like what Prettier does - use double quotes in jsx...

**Pull request merged**